projects/fal/src/fal/app.py

```diff
@@ -123,6 +124,24 @@ async def provide_hints_headers(request, call_next):
                 )
             return response
 
+        @app.middleware("http")
+        async def set_global_object_preference(request, call_next):
+            response = await call_next(request)
+            try:
+                from fal.toolkit.file.providers import fal
+
+                fal.GLOBAL_LIFECYCLE_PREFERENCE = request.headers[
+                    "X-Fal-Object-Lifecycle-Preference"
+                ]
+            except Exception:
+                from fastapi.logger import logger
+
+                logger.exception(
+                    "Failed set a global lifecycle preference %s",
+                    self.__class__.__name__,
+                )
+            return response
+
```

Comment:
Might be missing something, but I don't think this runtime patching would work retrospectively. The app is pickled during deployment and newer versions of fal are not present in the worker (unless you explicitly add fal to the requirements, but even then). If that was the intention here.